lib/parse/maven-graph-builder.ts

```diff
@@ -7,28 +7,46 @@ export class MavenGraphBuilder {
     this.#graph = {
       rootId,
       nodes: {
-        [rootId]: { dependsOn: [] },
+        [rootId]: { dependsOn: [], parents: [], reachesProdDep: false },
```

Comment:
Does this break or change any functionality for consumers outside of this module? Only asking since it's a public module and I am not sure of the blast radius.

Reply:
Nope, this part is purely an internal implementation detail

Reply:
Also, I would not consider this module public. The only real public facing interface is `inspect` from `index.ts`